tests/js/tests/prolog-and-literals.test.ts

```diff
@@ -818,6 +802,476 @@ describe("Prolog + Literals", () => {
+                    for (const recipe of oldRecipes) await recipe.delete();
+
+                    const source1 = Literal.from("Source 1").toUrl()
+                    const source2 = Literal.from("Source 2").toUrl()
+                    
+                    const recipe1 = new Recipe(perspective!, undefined, source1)
+                    recipe1.name = "Recipe 1: Name";
+                    await recipe1.save();
+
+                    const recipe2 = new Recipe(perspective!, undefined, source2)
+                    recipe2.name = "Recipe 2: Name";
+                    await recipe2.save();
+
+                    const recipe3 = new Recipe(perspective!, undefined, source2)
+                    recipe3.name = "Recipe 3: Name";
+                    await recipe3.save();
+
+                    const allRecipes = await Recipe.findAll(perspective!);
+                    expect(allRecipes.length).to.equal(3);
+
+                    const source1Recipes = await Recipe.findAll(perspective!, { source: source1 });
+                    expect(source1Recipes.length).to.equal(1);
+                    expect(source1Recipes[0].name).to.equal("Recipe 1: Name");
+
+                    const source2Recipes = await Recipe.findAll(perspective!, { source: source2 });
+                    expect(source2Recipes.length).to.equal(2);
+
+                    await recipe1.delete();
+                    await recipe2.delete();
+                    await recipe3.delete();
+                })
+
+                it("findAll() works with properties query", async () => {
+                    // Clear all previous recipes
+                    const allRecipes = await Recipe.findAll(perspective!);
+                    for (const recipe of allRecipes) await recipe.delete();
+                    
+                    let root = Literal.from("findAll test 1").toUrl()
+                    const recipe = new Recipe(perspective!, root);
+                    recipe.name = "recipe://test_name";
+                    recipe.booleanTest = true;
+                    await recipe.save();
+
+                    const me = await ad4m!.agent.me();
+
+                    // Test recipes with all properties
+                    const recipesWithAllAttributes = await Recipe.findAll(perspective!);
+                    expect(recipesWithAllAttributes[0].name).to.equal("recipe://test_name")
+                    expect(recipesWithAllAttributes[0].booleanTest).to.equal(true)
+                    expect(recipesWithAllAttributes[0].author).to.equal(me!.did)
+                    
+                    // Test recipes with name only
+                    const recipesWithNameOnly = await Recipe.findAll(perspective!, { properties: ["name"] });
+                    expect(recipesWithNameOnly[0].name).to.equal("recipe://test_name")
+                    expect(recipesWithNameOnly[0].booleanTest).to.be.undefined
+                    expect(recipesWithNameOnly[0].author).to.be.undefined
```

Comment:
I'm not sure about the utility of constraining it to certain properties, but I could fiture a case where we have big properties (especially if at some point we would have automatic loading/instantiation as default).. but even if we would have that, I think author and timestamp should always be there.